src/Features/CSharp/Portable/AssignOutParameters/AssignOutParametersCodeFixProvider.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.AssignOutParameters
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal class AssignOutParametersCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        private const string CS0177 = nameof(CS0177); // The out parameter 'x' must be assigned to before control leaves the current method
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(CS0177);
```

Comment:
Why not just string literal `"CS0177"`?

Reply:
standard IDE pattern to make it clear what each is. i'm just being consistent here.